src/wolff_producer.erl

```diff
@@ -478,13 +491,20 @@ ensure_delayed_reconnect(#{config := #{reconnect_delay_ms := Delay0},
                            topic := Topic,
                            partition := Partition,
                            reconnect_timer := ?no_timer
-                          } = St) ->
+                          } = St, DelayStrategy) ->
   Attempts = maps:get(reconnect_attempts, St, 0),
   Attempts > 0 andalso Attempts rem 10 =:= 0 andalso
     log_error(Topic, Partition, "still disconnected after ~p reconnect attempts", [Attempts]),
-  %% add an extra random delay to avoid all partition workers
-  %% try to reconnect around the same moment
-  Delay = Delay0 + rand:uniform(1000),
+  Delay =
+    case DelayStrategy of
+      no_delay_for_first_attempt when Attempts =:= 0 ->
+        %% this is the first attempt after disconnected, no delay
+        0;
```

Comment:
If the delay is 0, do I still need to print this log?

```
log_info(Topic, Partition, "will try to rediscover leader connection after ~p ms delay", [Delay]),
```

Reply:
yes. there will be a "lost connection" log, we should log the attempt to make it symmetrical